File: src/main/java/com/ms/semicolans/sereportapi/sereportapi/dto/requestdto/BankTransferDTO.java
package com.ms.semicolans.sereportapi.sereportapi.dto.requestdto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BankTransferDTO {
    private String referenceNo;
    private LocalDate createDate;
    private BigDecimal paidAmount;
    private BigDecimal serviceCharge;
    private String bankCode;
    private String bankName;
    private String branchCode;
    private String branchName;
    private String accountType;
    private String accountNo;
}

File: lib/model/bank_transfer_dto.dart
class BankTransferDTO {
  final String referenceNo;
  final DateTime createDate;
  final String paidAmount;
  final String serviceCharge;
  final String? bankCode;
  final String bankName;
  final String? branchCode;
  final String? branchName;
  final String? accountType;
  final String accountNo;

  BankTransferDTO({
    required this.referenceNo,
    required this.createDate,
    required this.paidAmount,
    required this.serviceCharge,
    this.bankCode,
    required this.bankName,
    this.branchCode,
    this.branchName,
    this.accountType,
    required this.accountNo,
  });

  Map<String, dynamic> toJson() => {
        'referenceNo': referenceNo,
        'createDate': createDate.toIso8601String().split('T')[0],
        'paidAmount': paidAmount,
        'serviceCharge': serviceCharge,
        if (bankCode != null) 'bankCode': bankCode,
        'bankName': bankName,
        if (branchCode != null) 'branchCode': branchCode,
        if (branchName != null) 'branchName': branchName,
        if (accountType != null) 'accountType': accountType,
        'accountNo': accountNo,
      };
}

